pypeit/scripts/ql.py

```diff
@@ -19,290 +19,519 @@
+    setups = []
+    calibs = []
+    for f in calib_files:
+        try:
+            setup, calib, detname = CalibFrame.parse_calib_key(
+                                        CalibFrame.parse_key_dir(str(f), from_filename=True)[0])
+        except:
+            continue
+        setups += [setup]
+        calibs += [calib]   # 'calib' is a comma-separated list of integers or 'all'
+
+    # Find the unique setups
+    setups = np.unique(setups)
+    if len(setups) != 1:
+        msgs.error(f'Calibration files for more than one setup found in {_calib_dir}, '
+                    'according to their file names.  Calibration directory should only hold data '
+                    'for *one* setup.')
+    setup = setups[0]
+
+    # Get the unique calibration groups identifiers
+    unique_calibs = np.unique(np.concatenate([c.split(',') for c in calibs])).tolist()
+    # Ignore any calibs assigned to all
+    if 'all' in unique_calibs:
+        unique_calibs.remove('all')
+    if len(unique_calibs) == 0:
+        # Everything assigned to all, so just use a default number
+        return setup, '0'
+    if len(unique_calibs) == 1:
+        # There's only one calibration group, so use it.
+        return setup, unique_calibs[0]
+
+    # TODO: For now, we fault here.  For quick-look, this means there should
+    # only be one calibration group, or all the calibrations are assigned to
+    # 'all' calibration groups.  Otherwise, we could have the user supply the
+    # calibration group.
+    msgs.error(f'Calibrations in {_calib_dir} are part of multiple calibration groups.  Unclear '
+               'how to proceed.')
 
-    return mtch[0], masters_dir
 
 class QL(scriptbase.ScriptBase):
 
     @classmethod
     def get_parser(cls, width=None):
-        parser = super().get_parser(description='Script to produce quick-look multislit PypeIt reductions', width=width)
+        parser = super().get_parser(description='Script to produce quick-look PypeIt reductions',
+                                    width=width)
         parser.add_argument('spectrograph', type=str,
                             help='A valid spectrograph identifier: {0}'.format(
                                  ', '.join(available_spectrographs)))
-        parser.add_argument('--rawfile_list', type=str, 
-                            help='File providing raw files to reduce including their path(s)')
-        parser.add_argument('--full_rawpath', type=str, 
-                            help='Full path to the raw files. Used with --rawfiles or --raw_extension')
-        parser.add_argument('--raw_extension', type=str, default='.fits',
-                            help='Extension for raw files in full_rawpath.  Only use if --rawfile_list and --rawfiles are not provided')
-        parser.add_argument('--rawfiles', type=str, nargs='+',
-                            help='space separated list of raw frames e.g. img1.fits img2.fits.  These must exist within --full_rawpath')
+
```

Comment:
My knee jerk reaction is that there are way too many arguments here. Let's discuss whether some can be consolidated, and/or if all the functionality they provide are really necessary. I have the feeling that the functionality has been overscoped. If this script is going to generate calibrations as well as perform reductions for multisit or echelle data, it is fair to ask why this approach is advantageous to a full run of pypeit?

Reply:
We talked through this, and removed some of the options.